### PR TITLE
Remove some old switches

### LINKED ISSUE #4311: Remove EnableJitInDiagMode from the code
It's an unused broken flag. We better remove it from the code and test cases. Long story short, if JIT is enabled, it's also enabled for Debugger and there is no another supported option.

### PR DESCRIPTION
**Pull request by rhuanjl**
Picking up from #6415 Remove some old switches from the code base. There are several more that could go but this seemed enough for a start.

1. Remove the EnableJitInDebugMode switch - this switch is from when the Jit was normally disabled if a debugger was attached - whereas the default behaviour has been for the Jit to be enabled during debugging for some time, hence remove the switch to Fix #4311 

2. Remove some old unused switches - these switches don't actually do anything, whatever features they are meant to control were either deleted or unguarded some time ago.

3. Remove several always on ES6 switches - these always on switches allowed disabling of parts of the Javascript spec that are settled and have been normal behaviour for a long time.

### CODE REVIEW

File: test/es6/rlexe.xml [hunk 1]
```diff
@@ -971,12 +969,6 @@
       <compile-flags>-es6spread</compile-flags>
     </default>
   </test>
-  <test>
-    <default>
-      <files>OS_917200.js</files>
-      <compile-flags>-ES6Destructuring-</compile-flags>
-    </default>
-  </test>
```
Comment (rhuanjl) [inline test/es6/rlexe.xml:979]: This test specifically tested behaviour when destructuring is disabled - with the switch removed this behaviour no longer exists, hence test removed.

**Comment by rhuanjl** [comment]
@zenparsing please could you take a look at this when you have a chance?

**Comment by rhuanjl** [comment]
@Fly-Style @penzn please could one of you review this - it's simple codebase quality stuff but would be nice to merge it.

**Review by ppenzin — APPROVED** [review]
I managed to overlook this PR and the other one somehow.

**Pull request merged**